src/ci/docker/host-x86_64/dist-x86_64-linux/build-gcc.sh

```diff
@@ -4,7 +4,8 @@ set -ex
 source shared.sh
 
 # Note: in the future when bumping to version 10.1.0, also take care of the sed block below.
-GCC=9.5.0
+# This version is specified in the Dockerfile
+GCC=$GCC_VERSION
```

Comment:
Let's add `u` to the bash modifier at the top of this script, to make debugging easier if the env. var. is unset.